daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -1014,36 +1009,30 @@ case class ParserTestCase(ptc: NodeSeq, parentArg: DFDLTestSuite)
 
     val testDataLength = lengthLimitInBits
 
-    val optExtVarDiag =
-      try {
-        processor = processor.withExternalDFDLVariables(externalVarBindings)
-        None
-      } catch {
-        case diag: Diagnostic =>
-          Some(diag)
-      }
-    if (optExtVarDiag.isDefined) {
-      toss(optExtVarDiag.get, implString)
-    } else {
-      val actual = processor.parse(new ByteArrayInputStream(testData), testDataLength)
-      val diagObjs = actual.getDiagnostics
+    try {
+      processor = processor.withExternalDFDLVariables(externalVarBindings)
+    } catch {
+      case e: Exception => toss(e, implString)
+    }
 
-      if (actual.isProcessingError) {
-        // Means there was an error, not just warnings.
-        if (diagObjs.length == 1) throw TDMLException(diagObjs.head, implString)
-        val diags = actual.getDiagnostics.map(_.getMessage()).mkString("\n")
-        throw TDMLException(diags, implString)
-      } else {
-        // If we think we've succeeded, verify there are no errors
-        // captured in the diagnostics. Otherwise there's probably
-        // an internal bug causing us to miss setting isProcessingError
-        val hasErrorDiags = diagObjs.exists { diag =>
-          diag.isError && !diag.isValidation
-        }
-        Assert.invariant(!hasErrorDiags)
+    val actual = processor.parse(new ByteArrayInputStream(testData), testDataLength)
+    val diagObjs = actual.getDiagnostics
+
+    if (actual.isProcessingError) {
+      // Means there was an error, not just warnings.
+      if (diagObjs.length == 1) throw TDMLException(diagObjs.head, implString)
```

Comment:
why toss above and throw here?

Reply:
This was existing code. I think I copied the toss from somewhere else. Not sure if there's a reason to use toss vs explicitly creating a TDMLException and throwing?

Reply:
The usual reason for toss is just for a breakpoint. So you can get the breakpoint before the context changes to the catcher. It's a "coding for debug" style.

Reply:
Okay, I'll switch it to a TDMLException then.